### ISSUE TITLE
setcap/setuid should be optional

### ISSUE THREAD

**Issue by Whissi** [iputils/iputils#175]
Commit 46d08f2080ad61b1d2fb2e5817fa04e5ca0db76b introduced a `setcap` or `setuid` call when installing iputils. This clashes with our (Gentoo) sandbox because build system is trying to modify files outside of `DESTDIR`:
```
>>> Install iputils-99999999 into /var/tmp/portage/net-misc/iputils-99999999/image/ category net-misc
ninja -v -j6 -l0 -C /var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999-build install
ninja: Entering directory `/var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999-build'
[1/14] /usr/lib/python-exec/python3.6/meson --internal vcstagger ../iputils-99999999/git-version.h.meson git-version.h s20190324 /var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999 @VCS_TAG@ '(.*)' git describe --dirty=+
[1/2] /usr/lib/python-exec/python3.6/meson install --no-rebuild
Installing ping to /var/tmp/portage/net-misc/iputils-99999999/image//usr/bin
Installing tracepath to /var/tmp/portage/net-misc/iputils-99999999/image//usr/bin
Installing traceroute6 to /var/tmp/portage/net-misc/iputils-99999999/image//usr/bin
Installing arping to /var/tmp/portage/net-misc/iputils-99999999/image//usr/bin
Running custom install script '/usr/lib/python-exec/python3.6/meson --internal gettext install --subdir=po --localedir=share/locale --pkgname=iputils'
Installing /var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999-build/po/ja.gmo to /var/tmp/portage/net-misc/iputils-99999999/image//usr/share/locale/ja/LC_MESSAGES/iputils.mo
Running custom install script '/var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999/build-aux/setcap-setuid.sh /usr/bin ping setuid /sbin/setcap'
/var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999/build-aux/setcap-setuid.sh: changing /usr/bin/ping to be setuid root executable
chown: cannot access '/usr/bin/ping': No such file or directory
chmod: cannot access '/usr/bin/ping': No such file or directory
Running custom install script '/var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999/build-aux/setcap-setuid.sh /usr/bin traceroute6 setuid /sbin/setcap'
/var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999/build-aux/setcap-setuid.sh: changing /usr/bin/traceroute6 to be setuid root executable
 * ACCESS DENIED:  fchownat:     /usr/bin/traceroute6
chown: changing ownership of '/usr/bin/traceroute6': Permission denied
 * ACCESS DENIED:  fchmodat:     /usr/bin/traceroute6
chmod: changing permissions of '/usr/bin/traceroute6': Permission denied
Running custom install script '/var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999/build-aux/setcap-setuid.sh /usr/bin arping setuid /sbin/setcap'
/var/tmp/portage/net-misc/iputils-99999999/work/iputils-99999999/build-aux/setcap-setuid.sh: changing /usr/bin/arping to be setuid root executable
chown: cannot access '/usr/bin/arping': No such file or directory
chmod: cannot access '/usr/bin/arping': No such file or directory
>>> Completed installing iputils-99999999 into /var/tmp/portage/net-misc/iputils-99999999/image/

 * Final size of build directory: 2204 KiB (2.1 MiB)
 * Final size of installed tree:   320 KiB

 * --------------------------- ACCESS VIOLATION SUMMARY ---------------------------
 * LOG FILE: "/var/log/sandbox/sandbox-4.log"
 *
VERSION 1.0
FORMAT: F - Function called
FORMAT: S - Access Status
FORMAT: P - Path as passed to function
FORMAT: A - Absolute Path (not canonical)
FORMAT: R - Canonical Path
FORMAT: C - Command Line

F: fchownat
S: deny
P: /usr/bin/traceroute6
A: /usr/bin/traceroute6
R: /usr/bin/traceroute
C: chown root /usr/bin/traceroute6

F: fchmodat
S: deny
P: /usr/bin/traceroute6
A: /usr/bin/traceroute6
R: /usr/bin/traceroute
C: chmod u+s /usr/bin/traceroute6
 * --------------------------------------------------------------------------------
```
Because we control things like that on our own, could you please hide that 'feature' behind an option which we can disable?

**Comment by kerolasa** [comment]
https://github.com/kerolasa/iputils/commit/d45a8c8034c076948cea2f37dee5f7930a50f8f0 implements `meson configure builddir -DNO_SETCAP_OR_SUID=true` but would that work for you? Even better if you could give that a try and inform if this is sufficient.

**Comment by Whissi** [comment]
Works for us.

**Comment by Whissi** [comment]
You mail also want to have a look at the following suggestion:
```diff
From 60ad30c6e38d2345580f083e67e1f54bd73be2ef Mon Sep 17 00:00:00 2001
From: Arfrever Frehtes Taifersar Arahesis <Arfrever@Apache.Org>
Date: Mon, 1 Apr 2019 00:30:20 +0000
Subject: [PATCH] setcap-setuid.sh: Respect ${DESTDIR}.

---
 build-aux/setcap-setuid.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/build-aux/setcap-setuid.sh b/build-aux/setcap-setuid.sh
index dba6c03..c93319c 100755
--- a/build-aux/setcap-setuid.sh
+++ b/build-aux/setcap-setuid.sh
@@ -6,6 +6,10 @@ exec_path="$1/$2"
 perm_type="$3"
 setcap="$4"
 
+if [ -n "${DESTDIR}" ]; then
+	exec_path="${DESTDIR%/}/${exec_path}"
+fi
+
 case "$perm_type" in
 	'caps')
 		echo "$0: calling: $setcap cap_net_raw+ep $exec_path"
-- 
2.21.0

```

**Comment by Arfrever** [comment]
+1 for respecting ${DESTDIR}.

**Comment by kerolasa** [comment]
That ${DESTDIR} fix is good idea, added to upstream in commit 784a67cb339cc43db86a676de00b51cb4101d1d6 with d45a8c8034c076948cea2f37dee5f7930a50f8f0 that I mentioned earlier. Thank you @Whissi for participation, and cheers Arfrever for patch.

**Comment by pevik** [comment]
NOTE: this needed further fix 473be64.
